File: C++/reverseStr.cpp
#include<iostream>
#include<string>

class Solution{
    public:
    std::string solve(std::string str) {
        int n = str.size();
        if(n<=1){return str;}
        int left=0;
        int right=n-1;
        while(left<right){
            int tmp=str[left];
            str[left]=str[right];
            str[right]=tmp;
            right--;
            left++;
        }
        return str;
    }
};

int main(){
    std::string str="abcd";
    Solution solution;
    std::cout<<"\""<<solution.solve(str)<<"\""<<std::endl;
    return 0;
}

File: python/reverseStr.py
class Solution:
    def solve(self, str: str) -> str:
        str_list = list(str)
        n = len(str_list)
        left = 0
        right = n - 1
        while left < right:
            str_list[left], str_list[right] = str_list[right], str_list[left]
            left += 1
            right -= 1
        return "".join(str_list)

s1="abcd"
s=Solution()
print(s.solve(s1))